src/cowboy_http2.erl

```diff
@@ -581,13 +580,9 @@ commands(State0=#state{socket=Socket, transport=Transport, http2_machine=HTTP2Ma
 	end,
 	commands(State, StreamID, Tail);
 commands(State=#state{socket=Socket, transport=Transport, http2_machine=HTTP2Machine0},
-		StreamID, [{flow, Size}|Tail]) ->
-	Transport:send(Socket, [
-		cow_http2:window_update(Size),
-		cow_http2:window_update(StreamID, Size)
-	]),
-	HTTP2Machine1 = cow_http2_machine:update_window(Size, HTTP2Machine0),
-	HTTP2Machine = cow_http2_machine:update_window(StreamID, Size, HTTP2Machine1),
+		StreamID, [{flow, _Size}|Tail]) ->
```

Comment:
We have to take the size into account. If the user requests for 8MB of data then we should set the window to 8MB directly instead of sending so many `window_update` frames.

Reply:
I think if the requested flow size is larger than our ideal window we should just send the flow size, otherwise we should use your algorithm in order to avoid sending too many `window_update` frames.

Reply:
Cowlib should probably have this size as argument and use it as part of the decision.

Reply:
OK. I agree with that.